pkg/cmd/roachtest/test_runner.go

```diff
@@ -688,6 +688,11 @@ func (r *testRunner) runWorker(
 			return err
 		}
 
+		if testToRun.spec.RequiresDeprecatedWorkload && workload[arch] == "" {
+			fmt.Fprintf(os.Stderr, "ERROR: unable to find required workload binary for %q", arch)
+			os.Exit(1)
```

Comment:
testRunner should never exit; shutdown logic involves tearing down running clusters. Technically, it should also never error out _unless_ it's an unrecoverable error, i.e., affects all tests. For now, you can use the same pattern as in the above `VerifyLibraries`. This will eventually be refactored into a (test) `Setup` step, whose failure would be localized, i.e., wouldn't fail the worker(s).